templates/snap-details.html

```diff
@@ -138,44 +138,55 @@ <h4>Where people are using {{ snap_title }}</h4>
       var ctrlDown = false;
 
       function keyDown(e) {
-        var key = e.key.toLowerCase();
-        if (key === 'control' || key === 'meta') {
-          ctrlDown = true;
-        }
-        if ((key === 'c' || key === 'x') && ctrlDown) {
-          if (typeof ga !== 'undefined') {
-            ga('gtm1.send', {
-              hitType: 'event',
-              eventCategory: 'Snap details',
-              eventAction: 'Copy CLI install instructions',
-              eventLabel: 'Copy {{ package_name }} input'
-            });
+        if (e && e.key) {
+          var key = e.key.toLowerCase();
+          if (key === 'control' || key === 'meta') {
+            ctrlDown = true;
+          }
+          if ((key === 'c' || key === 'x') && ctrlDown) {
+            if (typeof ga !== 'undefined') {
+              ga('gtm1.send', {
+                hitType: 'event',
+                eventCategory: 'Snap details',
+                eventAction: 'Copy CLI install instructions',
+                eventLabel: 'Copy {{ package_name }} input'
+              });
+            }
           }
         }
       }
 
       function keyUp(e) {
-        var key = e.key.toLowerCase();
-        if (key === 'control' || key === 'meta') {
-          ctrlDown = false;
+        if (e && e.key) {
+          var key = e.key.toLowerCase();
+          if (key === 'control' || key === 'meta') {
+            ctrlDown = false;
+          }
         }
       }
 
-      document.querySelector('#snap-install').addEventListener('keyup', keyUp);
-      document.querySelector('#snap-install').addEventListener('keydown', keyDown);
+      var snapInstall = document.querySelector('#snap-install');
+      if (snapInstall) {
+        snapInstall.addEventListener('keyup', keyUp);
+        snapInstall.addEventListener('keydown', keyDown);
+      }
 
-      var copy = new ClipboardJS('#copy');
+      if (ClipboardJS) {
+        var copy = new ClipboardJS('#copy');
+        
+        copy.on('success', function () {
+          if (typeof ga !== 'undefined') {
+            ga('gtm1.send', {
+              hitType: 'event',
+              eventCategory: 'Snap details',
+              eventAction: 'Copy CLI install instructions',
+              eventLabel: 'Copy {{ package_name }} button'
+            });
+          }
+        });
+      }
 
-      copy.on('success', function () {
-        if (typeof ga !== 'undefined') {
-          ga('gtm1.send', {
-            hitType: 'event',
-            eventCategory: 'Snap details',
-            eventAction: 'Copy CLI install instructions',
-            eventLabel: 'Copy {{ package_name }} button'
-          });
-        }
-      });
+      snapcraft.public.map('#js-snap-map', {{ countries|tojson }});
```

Comment:
This is a duplicated like.
We set up screenshots and map already in the beginning of this script.